### PR TITLE
Selenium fix

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:
follow up of #2494 

fixing code smells and disabling tests if docker can't be ran

Fixes

### CODE REVIEW

**Review by SudoBrendan — CHANGES_REQUESTED** [review]
So very close, would like to see separate error messages for localDevMode vs FailedDocker

File: test/e2e/setup.go [hunk 1]
```diff
@@ -75,15 +75,19 @@ var (
...
 	clients           *clientSet
+
+	dockerSucceeded bool
 )
 
 func skipIfNotInDevelopmentEnv() {
-	if !_env.IsLocalDevelopmentMode() {
+	// docker cmds will fail in INT until we figure out a solution since
+	// it is running from docker already
+	if !_env.IsLocalDevelopmentMode() || !dockerSucceeded {
```
[the remark this answers is not in the document]
Comment (facchettos) [reply test/e2e/setup.go]: I added a new function to skip only the portal tests when docker does not work

File: test/e2e/setup.go [hunk 2]
```diff
@@ -75,15 +75,25 @@ var (
 	vnetResourceGroup string
 	clusterName       string
 	clients           *clientSet
+
+	dockerSucceeded bool
 )
 
 func skipIfNotInDevelopmentEnv() {
+	// docker cmds will fail in INT until we figure out a solution since
```
Comment (cadenmarchese) [inline test/e2e/setup.go:83]: We can move this comment into the separate function.

**Review by cadenmarchese — CHANGES_REQUESTED** [review]

File: test/e2e/setup.go (hunk 2 above)
Comment (facchettos) [reply test/e2e/setup.go]: yes I messed up the refactoring haha, should be fine now

**Review by karanmagdani — APPROVED** [review]

**Review by petrkotas — APPROVED** [review]

**Pull request merged**